src/Microsoft.CodeQuality.Analyzers/UnitTests/Maintainability/CodeMetricsComputationTests.cs

```diff
@@ -0,0 +1,1796 @@
+
+        if (true)
+        {
+            var u1 = y1 - x1 * 1;
+            var u2 = y2 + x2 / 1;
+            var u3 = y3 > x3 % 1;
+            var u4 = y4 < x4 * 1;
+            var u5 = y5 * x5 * 1;
+            var u6 = y6 * x6 * 1;
+            var u7 = y7 * x7 * 1;
+            var u8 = y8 * x8 * 1;
+            var u9 = y9 * x9 * 1;
+            var u10 = y10 * x10 * 1;
+
+            var v1 = y1 - x1 * 1;
+            var v2 = y2 + x2 / 1;
+            var v3 = y3 > x3 % 1;
+            var v4 = y4 < x4 * 1;
+            var v5 = y5 * x5 * 1;
+            var v6 = y6 * x6 * 1;
+            var v7 = y7 * x7 * 1;
+            var v8 = y8 * x8 * 1;
+            var v9 = y9 * x9 * 1;
+            var v10 = y10 * x10 * 1;
+
+            var w1 = y1 - x1 * 1;
+            var w2 = y2 + x2 / 1;
+            var w3 = y3 > x3 % 1;
+            var w4 = y4 < x4 * 1;
+            var w5 = y5 * x5 * 1;
+            var w6 = y6 * x6 * 1;
+            var w7 = y7 * x7 * 1;
+            var w8 = y8 * x8 * 1;
+            var w9 = y9 * x9 * 1;
+            var w10 = y10 * x10 * 1;
+
+            var z1 = y1 - x1 * 1;
+            var z2 = y2 + x2 / 1;
+            var z3 = y3 > x3 % 1;
+            var z4 = y4 < x4 * 1;
+            var z5 = y5 * x5 * 1;
+            var z6 = y6 * x6 * 1;
+            var z7 = y7 * x7 * 1;
+            var z8 = y8 * x8 * 1;
+            var z9 = y9 * x9 * 1;
+            var z10 = y10 * x10 * 1;
+        }
+    }
+}
+";
+
+            var expectedMetricsText = @"
+Assembly: (Lines: 157, MntIndex: 27, CycCxty: 4, DepthInherit: 1)
+   C1: (Lines: 157, MntIndex: 27, CycCxty: 4, DepthInherit: 1)
+      C1.M(int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int): (Lines: 153, MntIndex: 27, CycCxty: 4)
+";
+
+            VerifyCSharp(source, expectedMetricsText);
+        }
+
+        [Fact]
+        public void VisualBasicTest01()
+        {
+            var source = @"
+Class C
+    Sub M(i As Integer)
+        Select Case i
+            Case 0
+                Exit Select
+            Case 1
+                Exit Select
+            Case Else
+        End Select
+    End Sub
+End Class
+";
+
+            var expectedMetricsText = @"
+Assembly: (Lines: 12, MntIndex: 92, CycCxty: 4, DepthInherit: 1)
+   C: (Lines: 12, MntIndex: 92, CycCxty: 4, DepthInherit: 1)
+      Public Sub M(i As Integer): (Lines: 9, MntIndex: 92, CycCxty: 4)
+";
+
+            VerifyBasic(source, expectedMetricsText);
+        }
+
+        [Fact]
+        public void VisualBasicTest02()
+        {
+            var source = @"
+Namespace N1
+    Class C
+        Sub M(Of T)(i As Integer)
+        End Sub
+
+        Class NestedClass(Of U)
+            Public Field0 As Integer = 0
+            Public Field1, Field2 As New Integer
+        End Class
+    End Class
+End Namespace
+
+Class TopLevel
+    Public ReadOnly Property P As Integer
+End Class
+
+";
+
+            var expectedMetricsText = @"
```

Comment:
are these new test or ported from VS side?

Reply:
Ported from VS side. Hopefully we will delete from VS side soon.